src/maybe.ts

```diff
@@ -143,6 +146,11 @@ export class Just<T> implements MaybeClasses<T> {
     return mapOrElse(orElseFn, mapFn, this);
   }
 
+  /** Method variant for [`Maybe.match`](../modules/_maybe_.html#match) */
+  match<T, U>(this: Maybe<T>, matchObj: { Just: (value: T) => U, Nothing: () => U }): U {
+    return mapOrElse(matchObj.Nothing, matchObj.Just, this);
```

Comment:
For most of the others, I've done this:

```ts
return match(matchObj, this);
```

That said, this implementation is a neat idea, and it's probably a good way to reimplement the static function itself!